generic_server.cc

```diff
@@ -211,6 +211,35 @@ future<> server::do_accepts(int which, bool keepalive, socket_address server_add
     });
 }
 
+shared_ptr<generic_server::connection>
+server::inject_connection(socket_address server_addr, connected_socket&& fd, socket_address addr) {
```

Comment:
Please, de-duplicate it with the same place in `server::do_accept()`'s right at once